tethys_gizmos/views/gizmos/jobs_table.py

```diff
@@ -1,23 +1,43 @@
+import inspect
 import logging
 import re
+from functools import wraps
 
 from django.http import JsonResponse
 from django.template.loader import render_to_string
 from tethys_compute.models import TethysJob, CondorWorkflow, DaskJob, DaskScheduler
 from tethys_gizmos.gizmo_options.jobs_table import JobsTable
 from tethys_sdk.gizmos import SelectInput
 from tethys_portal.optional_dependencies import optional_import
+from django.contrib.auth.decorators import login_required
+
 
 # optional imports
 server_document = optional_import("server_document", from_module="bokeh.embed")
 
 logger = logging.getLogger(f"tethys.{__name__}")
 
 
-def perform_action(request, job_id, action, success_message="", error_message=None):
+def async_login_required(func):
+    @wraps(func)
+    async def wrapper(request, *args, **kwargs):
+        redirect = login_required(lambda r: r)(request)
+        if redirect != request:
+            return redirect
+        return await func(request, *args, **kwargs)
+
+    return wrapper
+
+
+@async_login_required
+async def perform_action(
+    request, job_id, action, success_message="", error_message=None
+):
     try:
-        job = TethysJob.objects.get_subclass(id=job_id)
-        getattr(job, action)()
+        job = TethysJob.objects.get_subclass(id=job_id, user=request.user)
```

Comment:
What do you think about:
1. Default to this behavior for normal users.
2. Allow staff users to see any job (i.e. no user filter)
3. Creating a permission and/or permission group that would allow a normal user to see any job if they have the permission.